### PR TITLE
Completed HW 26.1

### PR DESCRIPTION
**Pull request by RomanMinevich**

### CODE REVIEW

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return item;
+    }
+
+    @Override
+    public List<Item> getAll() {
+        return null;
```
Comment (Sofasmile) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java]: add implementation

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Item;
+import mate.academy.internetshop.util.HibernateUtil;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ItemDaoHibernateImpl implements ItemDao {
+    private static final Logger log = Logger.getLogger(ItemDaoHibernateImpl.class);
```
Comment (Sofasmile) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:15]: do we use this logger here?

Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:15]: now we do)

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java (hunk 1 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java]: added, but there was no such task)
Снимок экрана 2019-10-09 в 16 03 24

Comment (boroda4436) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java]: @RomanMinevich thanks! Updated

File: pom.xml [hunk 3]
```diff
@@ -45,6 +45,13 @@
             <version>8.0.17</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+            <version>5.4.4.Final</version>
+        </dependency>
+
```
Comment (boroda4436) [inline pom.xml:53]: @RomanMinevich two empty lines are not required. Please remove one.

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Override
+    public Item create(Item item) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (boroda4436) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java]:
```suggestion
        Session session = null;
        try (session = HibernateUtil.getSessionFactory().openSession()) {
```

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            transaction = session.getTransaction();
+            Long id = (Long) session.save(item);
+            item.setId(id);
+            transaction.commit();
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error("Couldn't create an item");
+        }
```
Comment (boroda4436) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:36]:
```suggestion
        } finally {
            if (session != null) {
               session.close();
            }
        }
```

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.getTransaction();
+            session.delete(item);
+            transaction.commit();
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error("Couldn't delete an item with id " + id);
+        }
```
Comment (boroda4436) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:92]: add finally block and close session

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.getTransaction();
+            session.update(item);
+            transaction.commit();
+        } catch (Exception exception) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error("Couldn't update an item");
+        }
```
Comment (boroda4436) [inline src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:69]: add finally block and close session

File: src/main/java/mate.academy.internetshop/util/HibernateUtil.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package mate.academy.internetshop.util;
+
+import org.apache.log4j.Logger;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+public class HibernateUtil {
```
Comment (boroda4436) [inline src/main/java/mate.academy.internetshop/util/HibernateUtil.java:7]: @RomanMinevich Add private constructor

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java (hunk 4 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java]: Why can't I use try-with-resources statement here?

File: src/main/java/mate.academy.internetshop/util/HibernateUtil.java (hunk 8 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/util/HibernateUtil.java:7]: Ok, but for what reason?

File: pom.xml (hunk 3 above)
Comment (RomanMinevich) [reply pom.xml:53]: done

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java (hunk 5 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:36]: done

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java (hunk 7 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:69]: done

File: src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java (hunk 6 above)
Comment (RomanMinevich) [reply src/main/java/mate.academy.internetshop/dao/hibernate/ItemDaoHibernateImpl.java:92]: added

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**